src/ape/managers/chain.py

```diff
@@ -693,35 +696,45 @@ def get_deployments(self, contract_container: ContractContainer) -> List[Contrac
             List[:class:`~ape.contracts.ContractInstance`]: Returns a list of contracts that
             have been deployed.
         """
-        deployments_map = self._load_deployments_mapping()
-        network_name = self._network.name.replace("-fork", "")
-        ecosystem_name = self._network.ecosystem.name
+        contract_name = contract_container.contract_type.name
+        if not contract_name:
+            return []
 
         contract_addresses: List[AddressType] = []
-        if self._network.name == LOCAL_NETWORK_NAME or self._network.name.endswith("-fork"):
-            contract_addresses = (
-                self._local_deployments_mapping.get(ecosystem_name, {})
-                .get(network_name, {})
-                .get(contract_container.contract_type.name)
-                or []
+        deployments_map = (
+            self._local_deployments_mapping
+            if self._network.name == LOCAL_NETWORK_NAME or self._network.name.endswith("-fork")
+            else self._load_deployments_cache()
+        )
+        ecosystem_map = deployments_map.get(self._ecosystem_name, {})
+        network_map = ecosystem_map.get(self._data_network_name, {})
+        deployments = network_map.get(contract_name)
+        if not deployments:
+            return []
+
+        if isinstance(deployments[0], str):
+            # Handle previous schema of list of addresses instead of list of dicts.
```

Comment:
Probably add a TODO to deprecate this in like 0.6.x or something

Reply:
I also added it to my personal log of breaking changes I want (categorized by planned versions)

Reply:
resolved by https://github.com/ApeWorX/ape/pull/940/commits/901e8cf77ee5ea1f3496cb1d82ce473c97b8c6bd